lib-injection/auto_inject.rb

```diff
@@ -0,0 +1,84 @@
+return if ENV['skip_autoinject']
+
+begin
+  require 'open3'
+
+  _, status = Open3.capture2e({'skip_autoinject' => 'true'}, 'bundle show ddtrace')
+
+  if status.success?
+    puts '[DATADOG LIB INJECTION] ddtrace already installed... skipping auto-injection'
+    return
+  end
+
+  require 'bundler'
+  require 'shellwords'
+
+  if Bundler.frozen_bundle?
+    puts "[DATADOG LIB INJECTION] Cannot inject with frozen Bundler"
+    return
+  end
+
+  version = "<DD_TRACE_VERSION_TO_BE_REPLACED>"
+  sha = "<DD_TRACE_SHA_TO_BE_REPLACED>"
+
+  # Stronger restrict
+  condition = if !version.empty?
```

Comment:
I don't think `condition` express well what is storing. Maybe we could rename it to something, like `ddtrace_installation_path` 

```ruby
ddtrace_installation_path = if !version.empty?
  # For public release
  "bundle add ddtrace --version #{version.gsub(/^v/, '').shellescape} --require ddtrace/auto_instrument"
elsif !sha.empty?
  # For internal testing
  "bundle add ddtrace--github datadog/dd-trace-rb --ref #{sha.shellescape} --require ddtrace/auto_instrument"
end
```

Then on line 58, we could replace it with:

```ruby
output, status = Open3.capture2e(
  { 'skip_autoinject' => 'true', 'BUNDLE_GEMFILE' => datadog_gemfile.to_s },
  ddtrace_installation_path
)
```

What do you think?